auth/src/Repositories/UserRepository.php

```diff
@@ -0,0 +1,61 @@
+<?php namespace Miner\Auth\Repositories;
+
+use Miner\Auth\Entities\User;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class UserRepository
+{
+    public function __construct()
+    {
+        $this->db = app('db');
+    }
+
+    /**
+     * @param string $email
+     * @return User
+     *
+     * @throws \Exception
+     */
+    public function findByEmail(string $email): User
+    {
+        $result = $this->db->select('SELECT id, password_hash, role FROM users WHERE email = ?', [$email]);
+
+        if(!is_array($result) || count($result) === 0) {
```

Comment:
count is truthy, so you can just say `!count($result)` instead of `count($result) === 0`